service/orchestrator/cloud_service.go

```diff
@@ -91,40 +92,37 @@ func (s *Service) GetCloudService(_ context.Context, req *orchestrator.GetCloudS
 	return response, nil
 }
 
+// UpdateCloudService implements method for OrchestratorServer interface for updating a cloud service
+// TODO(all): remove ServiceId from request since it is accessible in service already
 func (s *Service) UpdateCloudService(_ context.Context, req *orchestrator.UpdateCloudServiceRequest) (response *orchestrator.CloudService, err error) {
-	if req.Service == nil {
-		return nil, status.Errorf(codes.InvalidArgument, "service is empty")
+	if err = req.Validate(); err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, err.Error())
 	}
 
-	if req.ServiceId == "" {
-		return nil, status.Errorf(codes.InvalidArgument, "service id is empty")
-	}
+	// Check if cloud service is in the DB
+	_, err = s.GetCloudService(context.Background(), &orchestrator.GetCloudServiceRequest{ServiceId: req.ServiceId})
```

Comment:
Not really necessary to call the complete getCloudservice function here. A simple Count() would be enough.